apps/autoscheduler/autotune_loop.sh

```diff
@@ -107,19 +119,43 @@ for ((i=$((FIRST+1));i<1000000;i++)); do
     # Copy the weights being used into the batch folder so that we can repro failures
     mkdir -p ${DIR}
     cp weights/* ${SAMPLES}/batch_${i}/
-    
+
     echo Compiling ${BATCH_SIZE} samples for batch_${i}...
     for ((b=0;b<${BATCH_SIZE};b++)); do
         S=$(printf "%d%02d" $i $b)
         FNAME=$(printf "%s_batch_%02d_sample_%02d" ${PIPELINE} $i $b)
-        make_sample "${DIR}/${b}" $S $FNAME &
+        make_sample_lib "${DIR}/${b}" $S $FNAME &
     done
     wait
 
+    echo Compiling ${BATCH_SIZE} stubs for batch_${i}...
+    for ((b=0;b<${BATCH_SIZE};b++)); do
+        S=$(printf "%d%02d" $i $b)
+        FNAME=$(printf "%s_batch_%02d_sample_%02d" ${PIPELINE} $i $b)
+        make_sample_stubs "${DIR}/${b}" $FNAME &
+    done
+    wait
+
+    STUBS=
+    for f in `ls ${DIR}/*/*.rungenstubs.o`; do
+        ALWAYS=`alwayslink $f`
+        STUBS="${STUBS} ${ALWAYS}"
+    done
+
+    echo Linking batch_${i}...
+    c++ -std=c++11 \
+        ../../tools/RunGenMain.cpp \
+        -I ../../include \
+        ${STUBS} \
+        ${DIR}/*/*.a \
```

Comment:
I guess this line only collects the successfully-compiled samples?

Reply:
Yes, but it would fail to link, since the RunGenStubs are basically never going to fail to generate.